Kernel fails to build

Hi!

I've encountered a kernel compile error when trying to build a system for Orange Pi 5 Plus. I'm using the following `flake.nix` (based on #57):

```nix
{
  description = "NixOS configuration";

  inputs = {
    unstable.url = "nixpkgs/nixos-unstable";
    nixpkgs.url = "nixpkgs/nixos-24.05";
    nixos-rk3588.url = "github:ryan4yin/nixos-rk3588";
  };

  outputs = { nixpkgs, nixos-rk3588, ... }: let
    # using the same nixpkgs as nixos-rk3588 to utilize the cross-compilation cache.
    inherit (nixos-rk3588.inputs) nixpkgs;
    system = "aarch64-linux";
  in {
    nixosConfigurations = {
      opi5plus = nixpkgs.lib.nixosSystem {
        inherit system;
        specialArgs = {
          rk3588 = {
            inherit nixpkgs;
            pkgsKernel = import nixpkgs {
              inherit system;
            };
          };
        };

        modules = [
          nixos-rk3588.nixosModules.boards.orangepi5plus.core
          ./system/configuration.nix
          ./system/hardware-configuration.nix
        ];
      };
    };
  };
}
```

The configuration.nix and hardware-configuration.nix are the files that were generated by nixos-generate-config while following the steps from [here](https://github.com/ryan4yin/nixos-rk3588/blob/main/UEFI.md#user-content-3-install-nixos-into-ssd--emmc-via-nixos-install).

I've partitioned my SSD and mounted to `/mnt` as Nix manual suggests, and ran this command (with no `flake.lock`, so it would generate one):

```sh
nixos-install --flake /mnt/etc/nixos#opi5plus --root /mnt -v
```
The command starts with (there are some cached things, since I'm re-running it to record the logs):

```console
warning: creating lock file '/mnt/etc/nixos/flake.lock'
building the flake in path:/mnt/etc/nixos?lastModified=1729524129&narHash=sha256-JfDZbS9aPugcmOWCuuPm9tHk9uX7m8EUX6xDMsMIYFE%3D...
copying path '/nix/store/6k9931glp8brpwrqnvgqihaca5bvja5v-source' to 'local'...
[1 copied (13.9 MiB)] evaluating derivation 'path:/mnt/etc/nixos?lastModified=17

these 6 derivations will be built:
  /nix/store/17l8mwjcaaa0a0l68kq6w1id48yc1r03-k.drv
  /nix/store/1fnvp6cgw1h1ri6nr0j35gvr366jcl3l-k-modules.drv
  /nix/store/8ffm72irgjiga2zkniy77zx64fwaia6h-k-modules-shrunk.drv
  /nix/store/6n4kh18al4k0gqpbgvx0h43nvqzpxvgv-initrd-k.drv
  /nix/store/sym2igl1manb05p2jg39495vgbb16sgr-boot.json.drv
  /nix/store/8ag5gfzx1w5rms4km3vkgripy5sz0jlx-nixos-system-nixos-24.05.20241009.d51c286.drv
building '/nix/store/17l8mwjcaaa0a0l68kq6w1id48yc1r03-k.drv'...
```

Then it prints CC/LD/... messages about building the kernel modules, until it fails (some 20-30 minutes later) with this:

```console
k>   CC [M]  drivers/hwmon/w83l785ts.o
k>   CC [M]  drivers/hwmon/w83l786ng.o
k>   LD [M]  drivers/hwmon/nct6775.o
k>   AR      drivers/hwmon/built-in.a
k> /nix/store/f3l6xjrp2yf76crhqfgyl6645skmyjl8-binutils-2.41/bin/ld: warning: -z
 relro ignored
k> make[1]: *** [../scripts/Makefile.build:500: drivers] Error 2
k> make: *** [../Makefile:2012: .] Error 2
error: builder for '/nix/store/17l8mwjcaaa0a0l68kq6w1id48yc1r03-k.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/8ag5gfzx1w5rms4km3vkgripy5sz0jlx-nixos-system-nixos-24.05.20241009.d51c286.drv' failed to build
[root@orangepi5plus:/mnt/etc/nixos]#
```

Can anyone suggest what could be the next step to debug this? I've been unsuccessfully googling for a way to get the full Make error message from `nixos-rebuild`, but can't find one.

--- a/flake.nix
+++ b/flake.nix
@@ -2,7 +2,7 @@
   description = "A minimal NixOS configuration for the RK3588/RK3588S based SBCs";
 
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
     flake-utils.url = "github:numtide/flake-utils";
 
     nixos-generators = {

